cache/cache.go

```diff
@@ -24,7 +24,7 @@ type ContentMetadata struct {
 
 type CachedData struct {
 	ContentMetadata
-	Data io.Reader
+	Data io.ReadCloser
```

Comment:
I think it'd be worth to leave a comment here why a client of this object is responsible for closing the reader